src/realm/obj.cpp

```diff
@@ -1935,6 +1935,60 @@ bool Obj::remove_backlink(ColKey col_key, ObjLink old_link, CascadeState& state)
 }
 
 void Obj::assign(const Obj& other)
+{
+    copy_other_object(other);
+    auto copy_links = [this, &other](ColKey col) {
+        auto t = m_table->get_opposite_table(col);
+        auto c = m_table->get_opposite_column(col);
+        auto backlinks = other.get_all_backlinks(col);
+        for (auto bl : backlinks) {
+            auto linking_obj = t->get_object(bl);
+            if (c.get_type() == col_type_Link) {
+                // Single link
+                REALM_ASSERT(!linking_obj.get<ObjKey>(c) || linking_obj.get<ObjKey>(c) == other.get_key());
+                linking_obj.set(c, get_key());
+            }
+            else {
+                auto l = linking_obj.get_linklist(c);
+                auto n = l.find_first(other.get_key());
+                REALM_ASSERT(n != realm::npos);
+                l.set(n, get_key());
+            }
+        }
+        return false;
+    };
+    m_table->for_each_backlink_column(copy_links);
+}
+
+void Obj::dup_and_handle_multiple_backlinks()
+{
+    auto copy_links = [this](ColKey col) {
+        auto t = m_table->get_opposite_table(col);
+        auto c = m_table->get_opposite_column(col);
+        auto backlinks = get_all_backlinks(col);
+
+        for (auto bl : backlinks) {
+            auto obj = m_table->create_object();
```

Comment:
this throws if the table has a primary key, can you add some tests around migrating a table with a primary key to embedded?

Reply:
I don't think we will ever reach the ```post_migration``` phase if there is a TopLevel table that has a primary key, and we try to turn it into an embedded table. 
We fail before while we validate the schema.

```
SECTION("change table to embedded - table has primary key") {
            Schema schema = {
                {"child_table",
                 {
                     {"value", PropertyType::Int, Property::IsPrimary{true}},
                 }},
                {"parent_table",
                 {
                     {"child_property", PropertyType::Object | PropertyType::Nullable, "child_table"},
                 }},
            };
            auto realm = Realm::get_shared_realm(config);
            realm->update_schema(schema, 1);
            auto child_table = ObjectStore::table_for_object_type(realm->read_group(), "child_table");
            REQUIRE_FALSE(child_table->is_embedded());

            REQUIRE_THROWS(
                realm->update_schema(set_table_type(schema, "child_table", ObjectType::Embedded), 2, nullptr));
        }
```